web/custom-plugins/forecast-of-response/src/components/Comments.tsx

```diff
@@ -128,17 +129,33 @@ export const CommentInput = (props: {
       message: message,
     }
     setLoading(true)
-    dmssAPI.generatedDmssApi
-      .explorerAdd({
-        dataSourceId: DEFAULT_DATASOURCE_ID,
-        dottedId: `${operationId}.comments`,
-        body: newComment,
-      })
-      .then(() => {
-        handleNewComment(newComment)
-        setMessage('')
+
+    //upload the comment entity to the comment package
+    addToPath(newComment, token, [], DEFAULT_DATASOURCE_ID, CommentPackage)
+      .then((uid: string) => {
+        const referenceObject = {
+          name: newComment.name,
+          id: uid,
+          type: newComment.type,
+        }
+        const dottedID = `${operationId}.comments`
+
+        //insert a reference to the operation's comments list.
+        dmssAPI
+          .insertDocumentReference({
+            dataSourceId: DEFAULT_DATASOURCE_ID,
+            documentDottedId: dottedID,
+            reference: referenceObject,
+          })
+          .then(() => {
+            handleNewComment(newComment)
+            setMessage('')
+          })
+          .finally(() => {
+            setLoading(false)
+          })
       })
-      .finally(() => {
+      .catch(() => {
```

Comment:
for some reason i was not able to adda .finally() here, I got an error message: "finally() does not exist on type Promise